modules/library/coverage/src/test/java/org/geotools/image/ImageWorkerTest.java

```diff
@@ -1694,7 +1695,39 @@ public void testMosaicBackgroundColor() {
         Object roiProperty = mosaicked.getProperty("ROI");
         assertThat(roiProperty, not((instanceOf(ROI.class))));
     }
-    
+
+    @Test
+    public void testPrepareRenderingAlphaOnNoData() {
+        BufferedImage red = getSyntheticRGB(Color.RED);
+        final int w = red.getWidth();
+        final int h = red.getHeight();
+
+        // Setting some image pixels to all zeros
```

Comment:
The test seems to be a bit weak, what about a combination of pixels that are zero only in one band, and/or in two bands,  and then in three bands, and making sure that only those with 3 zeroes have alpha set?